tests/unit/set.cpp

```diff
@@ -250,27 +315,35 @@ static grb::RC dense_tests(
 
 	std::cerr << "\b 16:";
 	ret = grb::set( dst, 0 );
+	ret = ret ? ret : grb::wait( dst );
 	ret = ret ? ret : grb::set< descriptors::dense >( dst, one_mask, src );
+	ret = ret ? ret : grb::wait( dst );
 	if( ret != ILLEGAL ) {
 		std::cerr << " expected ILLEGAL, got " << toString( ret ) << "\n";
 		return FAILED;
 	}
 	ret = SUCCESS;
-	if( nnz( dst ) != size( dst ) ) {
-		std::cerr << " expected " << size( dst ) << ", got " << nnz( dst ) << "\n";
-		ret = FAILED;
-	}
-	for( const auto &pair : dst ) {
-		if( pair.second != 0 ) {
-			std::cerr << "\t got ( " << pair.first << ", " << pair.second << " ), "
-				<< "expected zero values\n";
+
+	if( nonblocking_execution ) {
+		ret = grb::set( dst, 0 );
+	} else {
+		if( nnz( dst ) != size( dst ) ) {
+			std::cerr << " expected " << size( dst ) << ", got " << nnz( dst ) << "\n";
 			ret = FAILED;
 		}
+		for( const auto &pair : dst ) {
+			if( pair.second != 0 ) {
+				std::cerr << "\t got ( " << pair.first << ", " << pair.second << " ), "
+					<< "expected zero values\n";
+				ret = FAILED;
+			}
+		}
+		if( ret != SUCCESS ) { return ret; }
 	}
-	if( ret != SUCCESS ) { return ret; }
```

Comment:
Should this statement not remain outside of the else-if? (Consider e.g. if the nonblocking set dst back to 0 failed?)